packages/flutter/lib/src/material/range_slider.dart

```diff
@@ -1876,68 +2270,210 @@ class _RenderRangeSlider extends RenderBox with RelayoutWhenSystemFontsChangeMix
+    super.describeSemanticsConfiguration(config);
+    config.value = _resolveSemanticValue();
+    config.increasedValue = _resolveSemanticIncreasedValue();
+    config.decreasedValue = _resolveSemanticDecreasedValue();
+    if (isEnabled) {
+      config.onIncrease = _increaseAction;
+      config.onDecrease = _decreaseAction;
+    }
+    config.textDirection = textDirection;
+    config.isEnabled = isEnabled;
+    config.isSlider = true;
   }
+}
+
+class _FocusableThumb extends StatefulWidget {
+  const _FocusableThumb({
+    required this.values,
+    required this.divisions,
+    required this.focusNode,
+    required this.sliderState,
+    required this.thumb,
+    required this.sliderTheme,
+    required this.onChanged,
+    required this.renderSliderKey,
+    required this.targetPlatform,
+    required this.semanticFormatterCallback,
+    required this.renderThumbKey,
+  });
+
+  final RangeValues values;
+  final int? divisions;
+  final FocusNode focusNode;
+  final _RangeSliderState sliderState;
+  final Thumb thumb;
+  final SliderThemeData sliderTheme;
+  final ValueChanged<RangeValues>? onChanged;
+  final GlobalKey renderSliderKey;
+  final TargetPlatform targetPlatform;
+  final SemanticFormatterCallback? semanticFormatterCallback;
+  final GlobalKey renderThumbKey;
 
   @override
-  void detach() {
-    _valueIndicatorAnimation.removeListener(markNeedsPaint);
-    _state.startPositionController.removeListener(markNeedsPaint);
-    _state.endPositionController.removeListener(markNeedsPaint);
-    super.detach();
-  }
+  State<_FocusableThumb> createState() => _FocusableThumbState();
+}
 
+class _FocusableThumbState extends State<_FocusableThumb> {
   @override
-  void paint(PaintingContext context, Offset offset) {
-    _state.paintBottomValueIndicator?.call(context, offset);
-    _state.paintTopValueIndicator?.call(context, offset);
+  void initState() {
+    super.initState();
+    _actionMap = <Type, Action<Intent>>{
+      _AdjustSliderIntent: CallbackAction<_AdjustSliderIntent>(onInvoke: _actionHandler),
+    };
   }
 
-  @override
-  Size computeDryLayout(BoxConstraints constraints) {
-    return constraints.smallest;
+  _RenderThumb? get _renderThumbWidget =>
+      widget.renderThumbKey.currentContext?.findRenderObject() as _RenderThumb?;
+
+  void _actionHandler(_AdjustSliderIntent intent) {
+    final TextDirection directionality = Directionality.of(context);
+    final bool shouldIncrease = switch (intent.type) {
+      _SliderAdjustmentType.up => true,
+      _SliderAdjustmentType.down => false,
+      _SliderAdjustmentType.left => directionality == TextDirection.rtl,
+      _SliderAdjustmentType.right => directionality == TextDirection.ltr,
+    };
+
+    if (shouldIncrease) {
+      final String? semanticValuesInc = _renderThumbWidget?._resolveSemanticIncreasedValue();
+      _renderThumbWidget?._increaseAction();
+      if (semanticValuesInc != null) {
+        SemanticsService.announce(
```

Comment:
usually using announce is a red flag, do you know whats the native behavior?

Reply:
also if we have to manually create the string for announcement, this needs to go through localization